### DISCUSSION TITLE
Bulk remove file info

### DISCUSSION THREAD

**Discussion by nokatt** [stashapp/stash#4367] [Q&A] [answered]
Hi, sometimes scenes get falsely identified and there doesn't seem to be an option to mass delete the wrong cover, title, description and so from these scenes. Seems like I have to use sqlite to do this, but I'm not quite sure how to accomplish this, since they all over different tables. Is there a query that allows me to gather all the scenes from a particular performer or path and delete all their file infos?

**Comment by dingdongsolong4** [comment, marked as answer]
You are correct, there is currently no way to do this from the UI.

The easiest way is probably just to delete the scenes without deleting their files from disk and then rescan. If you don't want to regenerate any files after the rescan, you can also deselect the "Delete generated supporting files" option.

If you don't want to do that, then you can directly edit the database. It would be quite complicated, as there are indeed a few tables involved, but it is possible. And if you wanted to delete cover images as well, you'd definitely need an external script since the actual image data is most likely stored outside of the database (in the blobs folder), and thus can't be deleted using only SQL.

I think a good solution would be a button to "wipe all metadata", as described in #3878, which basically describes this exact issue.

**Comment by nokatt** [comment]
Thanks for your answer. 

I used the following query to reset every scene of a certain performer, but then I've seen your suggestion. If I had continued I would have to find a way to generate the blobs myself and write their names to the database which would be overkill. 
```
UPDATE scenes SET title=NULL, details=NULL, date=NULL, studio_id=NULL, organized=0, code=NULL, director=NULL WHERE id in (select scene_id from performers_scenes where performer_id = 4);
DELETE FROM scene_urls WHERE scene_id in (select scene_id from performers_scenes where performer_id = 4);
```

Yes deleting and rescanning seems to be the best solution, but sadly it doesn't really solve the issue itself. I had to create an extra tag to be able to exclude these scenes from the identifier in the future.

**Comment by dingdongsolong4** [comment]
FYI, you can use the Organized flag to exclude scenes from Identify and Auto Tag - scenes marked as Organized are skipped. I'm not exactly sure how you're using a tag for this, Identify doesn't let you exclude scenes based on tags as far as I know.

**Comment by nokatt** [comment]
> FYI, you can use the Organized flag to exclude scenes from Identify and Auto Tag - scenes marked as Organized are skipped. I'm not exactly sure how you're using a tag for this, Identify doesn't let you exclude scenes based on tags as far as I know.

Oh that's even better, I'm going to remove that tag again.
I thought this was the option to exclude a tag in the identifier. 
![image](https://github.com/stashapp/stash/assets/46605332/bc90b5db-b80a-4513-a409-fb1708b03967)

**Comment by dingdongsolong4** [comment]
Unfortunately no that isn't what that option does.

The Identify task runs through every scene in your database (excluding those marked as "Organized"), and tries to find a "match" for every scene in the Sources you specify. If it finds a match, then it copies all the metadata (name, date, etc) from that match and sets it on the scene.

Normally, if one source returns multiple matches, Identify will just take the first match and ignore the rest. This can cause issues though, because sometimes the second match is what you want, and the first match is completely incorrect.

So the `Skip matches that have more than one result` was added to solve this - if you enable that, then if multiple matches are found, Identify will simply skip the scene and set no metadata at all.

If you then _also_ set `Tag skipped matches with`, then Identify will add that tag to scenes that it skipped because it found multiple matches. The point of this is to allow you to then afterwards go through those scenes using i.e. the Scene Tagger and manually choose the correct match.
